mmdet3d/models/necks/dla_neck.py

```diff
@@ -0,0 +1,193 @@
+                out_channel,
+                out_channel,
+                up_kernel_size * 2,
+                stride=up_kernel_size,
+                padding=up_kernel_size // 2,
+                output_padding=0,
+                groups=out_channel,
+                bias=False)
+            setattr(self, 'proj_' + str(i), proj)
+            setattr(self, 'up_' + str(i), up)
+            setattr(self, 'node_' + str(i), node)
+
+    def forward(self, layers, start_level, end_level):
+        """Forward function.
+
+        Args:
+            layers (list[torch.Tensor]): features from multiple layers.
+            start_level (int): start layer for feature upsampling
+            end_level (int): end layer for feature upsampling
+        """
+        for i in range(start_level + 1, end_level):
+            upsample = getattr(self, 'up_' + str(i - start_level))
+            project = getattr(self, 'proj_' + str(i - start_level))
+            layers[i] = upsample(project(layers[i]))
+            node = getattr(self, 'node_' + str(i - start_level))
+            layers[i] = node(layers[i] + layers[i - 1])
+
+
+class DLAUp(nn.Module):
+    """DLAUp module for multiple layers feature extraction and fusion.
+
+    Args:
+        start_level (int): the start layer.
+        channels (List[int]): List of input channels of multi-scale
+                            feature map.
+        scales(List[int]): List of scale of different layers' feature.
+        in_channels (bool): List of input channels of different scales.
+                            Default: False.
+        norm_cfg (dict): Config dict for normalization layer. Default: None.
+    """
+
+    def __init__(self,
+                 start_level,
+                 channels,
+                 scales,
+                 in_channels=None,
+                 norm_cfg=None):
+        super(DLAUp, self).__init__()
+        self.start_level = start_level
+        if in_channels is None:
+            in_channels = channels
+        self.channels = channels
+        channels = list(channels)
+        scales = np.array(scales, dtype=int)
+        for i in range(len(channels) - 1):
+            j = -i - 2
+            setattr(
+                self, 'ida_{}'.format(i),
+                IDAUp(channels[j], in_channels[j:], scales[j:] // scales[j],
+                      norm_cfg))
+            scales[j + 1:] = scales[j]
+            in_channels[j + 1:] = [channels[j] for _ in channels[j + 1:]]
+
+    def forward(self, layers):
+        """Forward function.
+
+        Args:
+            layers (tuple[torch.Tensor]): features from multi-scale layers.
+        """
+        outs = [layers[-1]]
+        for i in range(len(layers) - self.start_level - 1):
+            ida = getattr(self, 'ida_{}'.format(i))
+            ida(layers, len(layers) - i - 2, len(layers))
+            outs.insert(0, layers[-1])
+        return outs
+
+
+@NECKS.register_module()
+class DLA_Neck(nn.Module):
```

Comment:
I think we can rename it to `DLANeck` (no `_` in class name). See [this class](https://github.com/open-mmlab/mmdetection3d/blob/9f0b01caf6aefed861ef4c3eb197c09362d26b32/mmdet3d/models/necks/imvoxel_neck.py#L8)